lib/ansible/modules/storage/netapp/na_ontap_autosupport.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/python
+"""
+create Autosupport module to enable, disable or modify
+"""
+
+# (c) 2018, NetApp, Inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+author: NetApp Ansible Team (ng-ansibleteam@netapp.com)
+description:
+  - "Enable/Disable Autosupport"
+extends_documentation_fragment:
+  - netapp.na_ontap
+module: na_ontap_autosupport
+options:
+  state:
+    description:
+      - Specifies whether the AutoSupport daemon is present or absent.
+      - When this setting is absent, delivery of all AutoSupport messages is turned off.
+    choices: ['present', 'absent']
+    default: present
+  node_name:
+    description:
+      - The name fo the filer that owns the AutoSupport Configuration.
+    required: true
+  transport:
+    description:
+      - The name of the transport protocol used to deliver AutoSupport messages
+    choices: ['http', 'https', 'smtp']
+  noteto:
+    description:
+      - Specifies up to five recipients of full AutoSupport e-mail messages.
+  post_url:
+    description:
+      - The URL used to deliver AutoSupport messages via HTTP POST
+  mail_hosts:
+    description:
+      - List of mail server(s) used to deliver AutoSupport messages via SMTP.
+      - Both host names and IP addresses may be used as valid input.
+      - One can optionally specify a username/password for authentication with the mail server
```

Comment:
1) What format would username/password take
2) Need to add `no_log` to the argument_spec if credentials can be supplied via this options

Reply:
1. We've removed the line. Mailhost is just a list of hostname and/or ip address.